File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.java
class Solution {
    private List<Integer> result = new ArrayList<>();

    private void traverse(TreeNode node) {
        if (node == null) {
            return;
        }
        traverse(node.left);
        result.add(node.val);
        traverse(node.right);
    }

    public List<Integer> inorderTraversal(TreeNode root) {
        traverse(root);
        return result;
    }
}

File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.py
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        def traverse(node):
            if node is None: return
            traverse(node.left)
            result.append(node.val)
            traverse(node.right)


        result = list()
        traverse(root)
        return result